BlueprintUI/Sources/ViewDescription/ViewDescription.swift

```diff
@@ -220,39 +217,40 @@ extension ViewDescription.Configuration {
     ///   be called on the next update.
     public subscript<Value>(keyPath: ReferenceWritableKeyPath<View, Value?>) -> Value? {
         get {
-            let key = AnyHashable(keyPath)
-            if let binding = bindings[key] as? ValueBinding<Value> {
+            if let binding = bindings[keyPath] as? ValueBinding<View, Value> {
                 return binding.value
             } else {
                 return nil
             }
         }
         set {
-            let key = AnyHashable(keyPath)
-            bindings[key] = ValueBinding(keyPath: keyPath, value: newValue)
+            bindings[keyPath] = ValueBinding(keyPath: keyPath, value: newValue)
         }
     }
     
 }
 
-extension ViewDescription.Configuration {
 
-    fileprivate class AnyValueBinding {
-        func apply(to view: View) { fatalError() }
-    }
+fileprivate protocol AnyValueBinding {
+    func apply(to view: UIView)
+}
 
-    fileprivate final class ValueBinding<Value>: AnyValueBinding {
-        var value: Value
+
+extension ViewDescription.Configuration {
+
+    fileprivate struct ValueBinding<View:UIView, Value> : AnyValueBinding {
+        
         var keyPath: ReferenceWritableKeyPath<View, Value>
+        var value: Value
 
         init(keyPath: ReferenceWritableKeyPath<View, Value>, value: Value) {
             self.value = value
             self.keyPath = keyPath
         }
 
-        override func apply(to view: View) {
+        func apply(to anyView: UIView) {
+            let view = anyView as! View
```

Comment:
Nit: Is there any way to avoid this cast? Could `AnyValueBinding` become a `struct` that's generic on the `View` perhaps?

Reply:
I might be able to get rid of the protocol via some clever closure stuff in init, let me see.

Reply:
Hmm, actually, no, that won't work, since we still need to access the typed value.

Reply:
Yeah I can't figure out how to do this.

Reply:
[This works](https://github.com/square/Blueprint/compare/kve/view-description-structs-only...westin/type-erasure?expand=1) but I'm not convinced it's much better...

Reply:
What you have seems fine!

Reply:
Yeah that's kinda where I had ended up – had to cast the value instead of the view; but one of them has to be cast. I think I'll stick with this